crates/eww/src/widgets/transform.rs

```diff
@@ -150,8 +175,9 @@ impl WidgetImpl for TransformPriv {
             };
 
             cr.scale(scale_x, scale_y);
+            cr.translate(transform_origin_x, transform_origin_y);
             cr.rotate(perc_to_rad(rotate));
-            cr.translate(translate_x, translate_y);
+            cr.translate(translate_x - transform_origin_x, translate_y - transform_origin_y);
```

Comment:
this doesn't break the order defined [here](https://github.com/elkowar/eww/blob/057297b4a339a4f72a6adf4ab587ebde2c331fcb/crates/eww/src/widgets/widget_definitions.rs#L1185), right?

Reply:
Well, even before this patch, the methods called on `cr` are in the order:

1. scale
2. rotate
3. translate

whereas in the comment you linked, it's claimed that the order is:

1. rotate
2. translate
3. scale

Sooooo... I don't know? Was it ever in the right order? Or is it sorted by the cairo-Context?

Reply:
it doesn't sort them, if i'm understanding [the docs](https://www.cairographics.org/manual/cairo-Transformations.html) right.
i suggest sorting them in the order described in the documentation, though i don't think it makes much of a difference.

also, the only thing causing the merge conflict is the changelog entry, that should be trivial to resolve when rebasing onto master ^^

Reply:
![Screenshot from 2024-08-29 17-46-53](https://github.com/user-attachments/assets/08406d75-2bb0-4a9e-a506-645e4c9c7bac)

There is a difference I'm afraid.
On the left side, scale is called after rotate + translate. basically, it warps the contents into a parallelogram when rotated. (the scaling is applied in the original directions of the x and y axis)
On the right is the same eww.yuck, but scale is called before rotate+translate, or the whole, scaled widget is rotated.

Generally speaking, users can always force a specific order by nesting multiple transforms. I did this here, where i nested the rotate+translate transform inside the scale transform:
![Screenshot from 2024-08-29 18-00-03](https://github.com/user-attachments/assets/7647be97-0776-4009-b8c5-b74ebd145b5a)
but it's not entirely the same; the bottom seems to be cut off (Because the size on the y-axis for this widget was limited by the scale to 60%)

soo... do we want the warping rotate as the default (scale after rotate, as documented), or do we want the non-warping rotate as the default (as previously implemented)?

Reply:
i'am pretty sure non-warping rotation is a saner default, the documentation should be updated accordingly